pangeo_forge_recipes/patterns.py

```diff
@@ -92,27 +94,13 @@ def __post_init__(self):
         assert self.index < self.sequence_len
 
 
-class Index(tuple):
-    """A tuple of ``DimIndex`` objects.
-    The order of the indexes doesn't matter for comparision."""
-
-    def __new__(self, args: Iterable[DimIndex]):
-        # This validation really slows things down because we call Index a lot!
-        # if not all((isinstance(a, DimIndex) for a in args)):
-        #     raise ValueError("All arguments must be DimIndex.")
-        # args_set = set(args)
-        # if len(set(args_set)) < len(tuple(args)):
-        #     raise ValueError("Duplicate argument detected.")
-        return tuple.__new__(Index, args)
-
-    def __str__(self):
-        return ",".join(str(dim) for dim in self)
-
-    def __eq__(self, other):
-        return (set(self) == set(other)) and (len(self) == len(other))
-
-    def __hash__(self):
-        return hash(frozenset(self))
+if TYPE_CHECKING:
+    # If we just do this, we can't initialize an Index by just writing
+    # Index(list_of_dims)
+    Index = FrozenSet[DimIndex]
+else:
+    # But if we just do this, it won't pass mypy 😖
+    Index = frozenset[DimIndex]
```

Comment:
Tests have shown that this does not work on python 3.8, even with `from __future__ import annotations`